### PR TITLE
chore(e2e): add `range.spec.ts`

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the current behaviour?

Part of https://github.com/taiga-family/taiga-ui/issues/4912

### CODE REVIEW

File: projects/demo-playwright/tests/kit/range/range.spec.ts [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+    test.describe('examples page', () => {
+        let example: Locator;
+        let po: TuiRangePO;
+
+        test.describe('change selected range on click', () => {
+            test.beforeEach(({page}) => {
+                example = page
+                    .locator('#base')
+                    .locator('[automation-id="tui-doc-example"]');
+                po = new TuiRangePO(example);
```
[the remark this answers is not in the document]
Comment (nsbarsukov) [reply projects/demo-playwright/tests/kit/range/range.spec.ts:16]:
```suggestion
                range = new TuiRangePO(example.locator('tui-range'));
```

File: projects/demo-playwright/utils/page-objects/range.po.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import {Locator} from '@playwright/test';
+
+export class TuiRangePO {
```
[the remark this answers is not in the document]
Comment (nsbarsukov) [reply projects/demo-playwright/utils/page-objects/range.po.ts:3]: **Usage:**

```ts
const range = new TuiRangePO(example.locator('tui-range'));

// [...]

await range.host.click({position: {x: 276, y: 0}});

await expect(range.left).toHaveValue('200');
await expect(range.right).toHaveValue('1000');
```

File: projects/demo-playwright/tests/kit/range/range.spec.ts [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+                });
+            });
+
+            test.describe('range with keySteps (from 0 to 1M) with 8 steps. Initial value [0, 100_000]', () => {
+                let output: Locator;
+
+                test.beforeEach(({page}) => {
+                    example = page
+                        .locator('#key-steps')
+                        .locator('[automation-id="tui-doc-example"]');
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/range/range.spec.ts:190]:
```suggestion
                    example = new TuiDocumentationPagePO(page).getExample('#key-steps');
```

File: projects/demo-playwright/tests/kit/range/range.spec.ts [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+                    output = example.locator('output code');
+                    po = new TuiRangePO(example);
+                });
+
+                test('ArrowUp increases value of the focused slider', async ({page}) => {
+                    await po.rightSlider().focus();
+                    expect(await output.innerText()).toEqual('[ 0, 100000 ]');
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/range/range.spec.ts:197]: Use [Auto-retrying assertion](https://playwright.dev/docs/test-assertions#auto-retrying-assertions):

```ts
await expect(output).toHaveText('[ 0, 100000 ]');
// or
await expect(output).toContainText('[ 0, 100000 ]')
```

Comment (splincode) [reply projects/demo-playwright/tests/kit/range/range.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/range/range.spec.ts [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+
+            test.describe('range with keySteps (from 0 to 1M) with 8 steps. Initial value [0, 100_000]', () => {
+                let output: Locator;
+
+                test.beforeEach(({page}) => {
+                    example = page
+                        .locator('#key-steps')
+                        .locator('[automation-id="tui-doc-example"]');
+                    output = example.locator('output code');
+                    po = new TuiRangePO(example);
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo-playwright/tests/kit/range/range.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/range/range.spec.ts (hunk 3 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/range/range.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/range/range.spec.ts [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+            });
+        });
+
+        test.describe('keyboard interactions', () => {
+            test.describe('basic range (from 0 to 1000 with 250 steps). Initial value [0, 250]', () => {
+                test.beforeEach(({page}) => {
+                    example = page
+                        .locator('#segments')
+                        .locator('[automation-id="tui-doc-example"]');
+                    po = new TuiRangePO(example);
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo-playwright/tests/kit/range/range.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/range/range.spec.ts [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+                await expect(example).toHaveScreenshot('03-range-click-checks-5-10.png');
+            });
+        });
+
+        test.describe('keyboard interactions', () => {
+            test.describe('basic range (from 0 to 1000 with 250 steps). Initial value [0, 250]', () => {
+                test.beforeEach(({page}) => {
+                    example = page
+                        .locator('#segments')
+                        .locator('[automation-id="tui-doc-example"]');
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo-playwright/tests/kit/range/range.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/range/range.spec.ts [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+
+    test.describe('examples page', () => {
+        let example: Locator;
+        let po: TuiRangePO;
+
+        test.describe('change selected range on click', () => {
+            test.beforeEach(({page}) => {
+                example = page
+                    .locator('#base')
+                    .locator('[automation-id="tui-doc-example"]');
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo-playwright/tests/kit/range/range.spec.ts]: fixed

File: projects/demo-playwright/utils/page-objects/range.po.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo-playwright/utils/page-objects/range.po.ts:3]: fixed

File: projects/demo-playwright/tests/kit/range/range.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo-playwright/tests/kit/range/range.spec.ts]: fixed

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**